doc/cookbook/source/examples/binary_classifier/averaged_perceptron.rst

```diff
@@ -0,0 +1,61 @@
+===================
+Averaged Perceptron
+===================
+
+The Averaged Perceptron is an online binary classifier which is used to determine whether an input,
+namely a feature vector :math:`\mathbf{x}`, belongs to some specific class or not. It is an extension
+of the simple perceptron algorithm because it uses the `average weight` vector
+and the `average bias` to classify the given inputs.
+The predicted class :math:`c` is given by the function below:
+
+.. math::
+
+    c = \theta\left(\mathbf{w} \cdot \mathbf{x}+b\right)
+
+Where :math:`\mathbf{w}` is the average weight vector, :math:`\mathbf{x}` is the input vector we want to classify,
+:math:`b` is the average bias and :math:`\theta` is the step function defined as:
+
+.. math::
+
+    \theta(x) =
+    \begin{cases}
+    1 & x > 0 \\
+    0 & x = 0 \\
+    -1 & x < 0
+    \end{cases}
+
+See Chapter 17 in :cite:`barber2012bayesian` for a brief explanation of the perceptron.
+
+-------
+Example
+-------
+
+Given a linearly separable dataset, we create some CDenseFeatures
+(RealFeatures, here 64 bit float values) and some :sgclass:`CBinaryLabels` to set up the training and validation sets.
+
+.. sgexample:: averaged_perceptron.sg:create_features
+
+Secondly, we create the :sgclass:`CAveragedPerceptron` instance by passing it the traning features and labels.
+We also set its learn rate and its maximum iterations.
+
+.. sgexample:: averaged_perceptron.sg:set_parameters
+
+Then we train the :sgclass:`CAveragedPerceptron` and we apply it to the test data, which gives the predicted :sgclass:`CLabels`.
+
+.. sgexample:: averaged_perceptron.sg:train_and_apply
+
+We can now also extract :math:`\mathbf{w}` and :math:`b`.
```

Comment:
We can also extract the average weight w and the bias b.

(reference the letters introduced above), shorten (no "now")